pkg/knuu/instance.go

```diff
@@ -285,28 +236,46 @@ func (i *Instance) ExecuteCommand(command ...string) (string, error) {
 		}
 		return output, nil
 	} else if i.IsInState(Started) {
-		pod, err := k8s.GetFirstPodFromStatefulSet(k8s.Namespace(), i.k8sName)
+		instanceName := i.k8sName
+		containerName := i.k8sName
+		if i.isSidecar {
+			instanceName = i.parentInstance.k8sName
+		}
+
+		pod, err := k8s.GetFirstPodFromStatefulSet(k8s.Namespace(), instanceName)
 		if err != nil {
 			return "", fmt.Errorf("error getting pod from statefulset '%s': %v", i.k8sName, err)
 		}
 		commandWithShell := []string{"/bin/sh", "-c", strings.Join(command, " ")}
-		output, err := k8s.RunCommandInPod(k8s.Namespace(), pod.Name, i.k8sName, commandWithShell)
+		output, err := k8s.RunCommandInPod(k8s.Namespace(), pod.Name, containerName, commandWithShell)
 		if err != nil {
-			return "", fmt.Errorf("error executing command '%s' in started instance '%s': %v", command, i.k8sName, err)
+			if i.isSidecar {
+				return "", fmt.Errorf("error executing command '%s' in sidecar '%s' of instance '%s': %v", command, i.k8sName, i.parentInstance.k8sName, err)
+			} else {
+				return "", fmt.Errorf("error executing command '%s' in instance '%s': %v", command, i.k8sName, err)
+			}
```

Comment:
I would make the Sidecar checker as a separate if statement outside of the parent container checker. 
Have 3 nested ifs is not readable.

Reply:
Do you mean implement a function `handleCommandError(err, command, i)`?

Another option is not obvious to me, as I want to return an error depending on if the instance the command execution threw an error is a sidecar or not.
But to simplify it, we could use the same error message for both.

Reply:
Having a func is another solution that sgtm.

You can check if the instance is the sidecar or instance first before assigning the error. The idea is to as explicit as possible due to the size of the struct. In addition to that, reducing the amount of nested statements 

For example: 
```go
output, err := k8s.RunCommandInPod(k8s.Namespace(), pod.Name, containerName, commandWithShell)
if i.isSidecar && err != nil {
    return "", fmt.Errorf("error executing command '%s' in sidecar '%s' of instance '%s': %v", command, i.k8sName, i.parentInstance.k8sName, err)
} else if !i.isSidecar && err != nil{
    return "", fmt.Errorf("error executing command '%s' in instance '%s': %v", command, i.k8sName, err)
}
```

Another can be regarding the whole else if statement where you first check for the isSidecar and then have the general logic
```go
var instanceName string
var errMsg err
containerName := i.k8sName

if isSidecar {
  instanceName = i.parentInstance.k8sName
  errMsg = fmt.Errorf("error executing command '%s' in sidecar '%s' of instance '%s': %v", command, i.k8sName, i.parentInstance.k8sName, err)
} else {
  // whatever opposite for instance
}
output, err := k8s.RunCommandInPod(k8s.Namespace(), pod.Name, containerName, commandWithShell)
if err != nil {
   return "", errMsg
}

return "", nil

```

Reply:
Thank you for the feedback! See https://github.com/celestiaorg/knuu/pull/168/commits/49bd378404e5111e1907835989e76da4bf1a94f5 and https://github.com/celestiaorg/knuu/pull/168/commits/a57b615786fc8180d6a2bf14fc9bd3c5cd50a681.

Now the code is much more readable :+1: